frontend/homework/11/src/component/Main.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react";
+import "../styles/Main.scss";
+import { APIQuote } from "../utils/ApiQuote";
+import Quote from "./Quote";
+import Filter from "./Filter";
+import axios from 'axios';
+
+function Main() {
+    let [quotes, setQuotes] = useState<APIQuote[]>([]);
+
+    const [filterArray, setFilterArray] = useState<string[]>([]);
+
+    function addAQuote() {
+        axios.get("https://api.quotable.io/quotes/random")
+            .then((response) => {
+                const data = response.data;
+                setQuotes([...data, ...quotes]);
+                console.log(quotes);
+            })
+            .catch((error) => {
+                console.error("Error fetching new quote:", error);
+            });
+    }
+
+
+    const handleDeleteFilter = (filterArray: string) => {
+        setFilterArray((prevFilters) => prevFilters.filter((f) => f !== filterArray));
+    };
+
+    useEffect(() => {
+        //this runs only once
+        axios.get("https://api.quotable.io/quotes/random?limit=3")
+            .then((response) => {
+                setQuotes(response.data);
+            });
+        console.log("fetched 3", quotes);
+    }, []);
+
+    const filteredQuotes = quotes.filter(
+        (quote) =>
+            filterArray.length === 0 ||
+            filterArray.some((tag) => quote.tags.includes(tag))
+    );
```

Comment:
This is okay but for a more complex component, see PresentationContainer Pattern in React